### ISSUE TITLE
iframe rule should not be limited to the same tree (4b1c6c)

### ISSUE THREAD

**Issue by WilcoFiers** [act-rules/act-rules.github.io#835]
In the rule "Iframes with identical accessible names serve equivalent purpose", applicability says the following:

> This rule applies to any set of any two or more iframe elements in the same document tree

The way this is formulated, causes iframes that are embedded each in their own shadow tree to be exempt from the rule. This seems odd, as there's no difference in how AT users will end up interacting with them. Even iframes embedded into the page through other iframes should not be exempt in my opinion.

**Comment by kasperisager** [comment]
Duplicate of #769

**Comment by Jym77** [comment]
@WilcoFiers Are you saying that AT do not respect iframe boundaries?
In relation to #768 I've benn trying the following document:
```html
<!DOCTYPE html>
<html>
<head><title>I have been framed!</title></head>

<body>
	<a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html">Contact us</a> all the time.

	<iframe srcdoc='<span><a href="/test-assets/links-with-identical-names-serve-equivalent-purpose-b20e66/about/contact.html">Contact us</a> from the iframe.</span>'></iframe>
</body>
</html>
```

With Chrome, the top-level accessibility tree only shows the first link ("all the time") and an `iframe` with no child and I need to manually go into the second document to see the second link ("from the iframe"). So, it seems that Chrome accessibility tree does not list all the links together but respects the iframe boundary.
This needs to be tested a bit more thoroughly.

**Comment by Jym77** [comment]
@WilcoFiers @kasperisager 
So, I'e been running some test with NVDA on Chrome. NVDA does not respect iframe boundaries and is dropping all the content upon seeing the page.

Which mean that, for #768 we likely want links to have different names *even if they are in an iframe*; and, for #769 (and this issue), we want iframes' ids to be unique *even if they are nested inside another iframe*.

Now, the correct applicability (for both rules) is (maybe?) links/iframe that are in the same flat tree. That should take care of iframe nesting. On the other hand, that seems weird that a dev has to take care of whatever their framework put in the components shadow tree on which they have no control…

**Comment by Jym77** [comment]
Closed by #982.
